lib/collection/request.js

```diff
@@ -107,7 +107,13 @@ _.assign(Request.prototype, /** @lends Request.prototype */ {
             /**
              * @type {Certificate|undefined}
              */
-            certificate: options.certificate && new Certificate(options.certificate)
+            certificate: options.certificate && new Certificate(options.certificate),
+
+            /**
+             * @private
+             * @type {Object}
+             */
+            size: options && options.size
```

Comment:
Please refer to `request.size()` as a function. Compute size when called. The non-computable size, IMHO, is body size and that should be returned by adding a .size property in request-body which in turn is used inside request object.

Places where we know the size only after reading, etc - can be stored as a fallback in a private property of the body.